Improve markdown list formatting in extension description

st-Wook opened the pull request:
For sublists, line breaks are needed at both the beginning and the end. However, currently, only a line break is being added at the beginning. 
 
1. Add a line break at the end of each sub-list
2. Skip adding a line break if there is already a duplicate line break
3. Correct the space indentation if it is less than or equal to 4 spaces for sub-items
 
 ```md
 Text:
 1. des1
 2. des2
 
 Text:
 * des1
   * subDes1
   * subDes2
 * des2
 ```
<https://wiki.gdevelop.io/gdevelop5/extensions/authorized-platforms-validation/#platforms-validation> (How to use)
<https://wiki.gdevelop.io/gdevelop5/extensions/sprite-sheet/> (How to use)

4ian commented:
> Should I add a feature to change the incorrect format from "-Text" to "- Text"? Or would it be better to directly change the extension description?

I guess for this it's better to issue a fix on the extension directly 👍

st-Wook commented:
I have made a PR by directly fixing a few incorrect list formats in the extension! ([PR](https://github.com/GDevelopApp/GDevelop-extensions/pull/904#issue-1741035647))

While fixing the format, I found that line breaks were not working in general description sentences. Thus, I have made additional changes so that '\n' is converted to '  \n'. (In fact, '\n' should not be there, but I added it for better readability when viewed in GitHub preview) ([Line Break Best Practices](https://www.markdownguide.org/basic-syntax/#line-break-best-practices)).

st-Wook commented:
It was my misunderstanding that line breaks had to be added at the beginning and end of nested lists.😅 I realized that the issue was not the line breaks but the formatting for the nested lists. Therefore, I have shifted the direction of this PR towards improving it in a different manner.

st-Wook commented:
I have switched to a manual validation approach, checking each line individually, as the previous regular expression approach had limitations. I have updated the changes in the PR description. Additionally, I have created an additional PR to address any necessary fixes found during the previous formatting fix PR. [Link](https://github.com/GDevelopApp/GDevelop-extensions/pull/905)

st-Wook commented:
It seems that line breaks aren't working in the editor extension description. Should I convert them manually in the script to match the format, or is it better to directly change the "\n" characters within the paragraph sections to two spaces?

st-Wook commented:
> I'm not sure to be honest. I think the editor follows the markdown convention which is that line breaks are not taken into account, except if you do two of them. Of course, it's a bit suprising (and GitHub flavored markdown don't do this, so it's surprising even for us who are regular markdown users) and controversial: https://gitlab.com/gitlab-org/gitlab/-/issues/199702
> 
> I think we should avoid generating line breaks if they are not rendered in the editor. The editor and the wiki should both adhere to the default markdown convention which is that single line breaks don't matter (except if you had two spaces at the end of a line, or if you use multiple line breaks).

Ah, I see. I have reverted the line break logic in the paragraph. What do you think about the other changes?
